src/core/utils/request/getRequestCookies.ts

```diff
@@ -37,7 +38,9 @@ export function getRequestCookies(request: Request): Record<string, string> {
   }
 }
 
+const cache = new WeakMap<Request, Record<string, string>>()
 export function getAllRequestCookies(request: Request): Record<string, string> {
```

Comment:
reading cookies isn't that expensive but it's idempotent to a single request, so we should be good to cache this